modules/private.php

```diff
@@ -0,0 +1,265 @@
+	<?php
+}
+
+/**
+ * Hides the blog's name on the login form for private blogs.
+ */
+function privatize_blog_maybe_mask_blog_name() {
+	if ( ! is_private_blog() ) {
+		return;
+	}
+
+	add_filter( 'bloginfo', 'privatize_blog_mask_blog_name', 3, 2 );
+}
+
+/**
+ * Replaces the the blog's "name" value with "Protected Blog"
+ *
+ * @see privatize_blog_maybe_mask_blog_name()
+ * @param mixed $value The requested non-URL site information.
+ * @param mixed $what  Type of information requested.
+ */
+function privatize_blog_mask_blog_name( $value, $what ) {
+	if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+		$value = __( 'Protected Blog', 'jetpack' );
+	}
+
+	return $value;
+}
+
+/**
+ * Changes a blog to private
+ */
+function make_blog_private() {
+	update_option( 'blog_public', -1 );
+}
+
+/**
+ * Changes a blog to public
+ */
+function make_blog_public() {
+	update_option( 'blog_public', 1 );
+}
+
+add_action( 'parse_request', 'privatize_blog', 100 );
+add_action( 'login_init', 'privatize_blog_maybe_mask_blog_name' );
+add_filter( 'preprocess_comment', 'privatize_blog_comments' );
+add_action( 'blog_privacy_selector', 'privatize_blog_priv_selector' );
+add_action( 'jetpack_activate_module_private', 'make_blog_private' );
+add_action( 'jetpack_deactivate_module_private', 'make_blog_public' );
+
+/**
+ * Don't let search engines index private sites
+ * or sites not deemed publicly available, like deleted, archived, spam.
+ *
+ * @param string $output Robots.txt output.
+ */
+function private_robots_txt( $output ) {
+	if ( ! is_publicly_available() ) {
+		$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+		$output .= "Disallow: /\n";
+	}
+
+	return $output;
+}
+add_filter( 'robots_txt', 'private_robots_txt', 10, 2 );
+
+/**
+ * Filters the link title attribute for the message displayed in the 'At a Glance' dashboard widget.
+ *
+ * @param string $text Default attribute text.
+ */
+function privatize_privacy_on_link_title( $text ) {
+	if ( '-1' === get_option( 'blog_public' ) ) {
+		return __( 'Your site is visible only to registered members', 'jetpack' );
+	}
+
+	return $text;
+}
+add_filter( 'privacy_on_link_title', 'privatize_privacy_on_link_title' );
+
+
+/**
+ * Filters the link label for the message displayed in the 'At a Glance' dashboard widget.
+ *
+ * @param string $text Default text.
+ */
+function privatize_privacy_on_link_text( $text ) {
+	if ( '-1' === get_option( 'blog_public' ) ) {
+		return __( 'Private', 'jetpack' );
+	}
+
+	return $text;
+}
+add_filter( 'privacy_on_link_text', 'privatize_privacy_on_link_text' );
+
+/**
+ * Output the meta tag that tells Pinterest not to allow users to pin
+ * content from this page.
+ * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+ */
+function private_no_pinning() {
+	echo '<meta name="pinterest" content="nopin" />';
+}
+add_action( 'wp_head', 'private_no_pinning' );
```

Comment:
In addition to this, should we do something about the sharing, Publicize, and Subscriptions features to make sure they can't get triggered when your site is set to Private?

Reply:
I _think_ Subscriptions will be blocked by `preprocess_comment` although I'm not sure how to test it.

Reply:
One way to make sure specific modules can't be used would be to disable them:
https://jetpack.com/support/module-overrides/#how-are-modules-disabled